mlflow/genai/scorers/builtin_scorers.py

```diff
@@ -715,6 +716,129 @@ def __call__(self, *, trace: Trace) -> list[Feedback]:
         return feedbacks
 
 
+def _validate_tool_calls_with_expectations(
+    actual_tools_called: list[ToolCallInfo],
+    expected_tool_calls: list[ToolCallInfo],
+    scorer_name: str,
+    check_order: bool = True,
+) -> Feedback:
+    """
+    Validate actual tool calls against expected tool calls.
+
+    Args:
+        actual_tools_called: List of actual ToolCallInfo objects from trace
+        expected_tool_calls: List of expected ToolCallInfo objects
+        scorer_name: Name of the scorer for feedback
+        check_order: Whether to validate order of tool calls (True) or just presence (False)
+
+    Returns:
+        Feedback with "yes" if validation passes, "no" if validation fails
+    """
+    if check_order:
+        if len(actual_tools_called) < len(expected_tool_calls):
+            return Feedback(
+                name=scorer_name,
+                value=CategoricalRating.NO,
+                rationale=(
+                    f"Expected {len(expected_tool_calls)} tool calls but only found "
+                    f"{len(actual_tools_called)} actual tool calls."
+                ),
+            )
+
+        # Check each expected tool call in order
```

Comment:
same here about using list equality instead - that way we can show all mismatches instead of one at a time